### PR TITLE
Docs for deferred constraints

### PR DESCRIPTION
**Pull request by ndeodhar**

### CODE REVIEW

File: docs/content/latest/api/ysql/commands/ddl_alter_table.md [hunk 1]
```diff
@@ -100,15 +100,40 @@ If enabled, row level security policies will be applied when the user is the tab
...
+Specify a table or column constraint.
 
 #### CONSTRAINT *constraint_name*
 
 Specify the name of the constraint.
 
-#### CHECK ( expression ) | FOREIGN KEY ( column_names ) *reference_clause*
+#### Foreign key
+
+`FOREIGN KEY` and `REFERENCES` specifies that the set of columns can only contain values that are present in the referenced column(s) of the referenced table. It is used to enfor
```
Comment (stevebang) [inline docs/content/latest/api/ysql/commands/ddl_alter_table.md:113]:
```suggestion
`FOREIGN KEY` and `REFERENCES` specify that the set of columns can only contain values that are present in the referenced columns of the referenced table. It is used to enfor
```

Comment (stevebang) [reply docs/content/latest/api/ysql/commands/ddl_alter_table.md:113]: Missing end of sentence too.

File: docs/content/latest/api/ysql/commands/ddl_alter_table.md [hunk 2]
```diff
@@ -100,15 +100,40 @@ If enabled, row level security policies will be applied when the user is the tab
...
+
+#### Unique
+
+This enforces that the set of columns specified in the `UNIQUE` constraint are unique in the table, that is, no two rows can have the same values for the set of columns specified
```
Comment (stevebang) [inline docs/content/latest/api/ysql/commands/ddl_alter_table.md:117]:
```suggestion
This enforces that the set of columns specified in the `UNIQUE` constraint are unique in the table, that is, no two rows can have the same values for the set of columns specified.
```

File: docs/content/latest/api/ysql/commands/ddl_alter_table.md [hunk 3]
```diff
@@ -100,15 +100,40 @@ If enabled, row level security policies will be applied when the user is the tab
...
+
+#### Check
+
+This is used to enforce that data in the specified table meets the requirements specified in the `CHECK` clause.
+
+#### Default
+
+This clause is used to specify a default value for the column. If an `INSERT` statement does not specify a value for the column, then the default value is used. If no default is
+
```
Comment (stevebang) [inline docs/content/latest/api/ysql/commands/ddl_alter_table.md:126]: Which clause is "This clause"? Also, end is incomplete.

File: docs/content/latest/api/ysql/commands/ddl_alter_table.md [hunk 4]
```diff
@@ -100,15 +100,40 @@ If enabled, row level security policies will be applied when the user is the tab
...
+#### Default
+
+This is used to specify a default value for the column. If an `INSERT` statement does not specify a value for the column, then the default value is used. If no default is specified for a column, then the default is NULL.
+
+#### Deferrable constraints
+
+Constraints can be deferred using the `DEFERRABLE` clause. Currently, only foreign key constraints
+can be deferred in YugabyteDB. A constraint that is not deferrable will be checked at the end of every
+statement. In the case of deferrable constraints, the checking of the constraint can be postponed
```
Comment (m-iancu) [inline docs/content/latest/api/ysql/commands/ddl_alter_table.md:131]: nit: technically it could be after each row within a statement.

Comment (ndeodhar) [reply docs/content/latest/api/ysql/commands/ddl_alter_table.md]: Done.

**Pull request merged**